src/RequestHeaderParser.php

```diff
@@ -163,6 +163,10 @@ private function parseRequest($headers)
             }
         }
 
+        if ($request->getProtocolVersion() === '1.1' && !$request->hasHeader('Host')) {
+            throw new \InvalidArgumentException('A client must send a host header field in all HTTP/1.1 request messages');
```

Comment:
The parameter to `parseRequest` (`$headers`) is not a valid argument.

Also, all exceptions in this method are `InvalidArgumentExceptions` so I was just keeping the format.